pyuvdata/uvcal/uvcal.py

```diff
@@ -4566,6 +5059,359 @@ def read_fhd_cal(
+                    n_settings = len(settings_file)
+                    settings_file_use = settings_file[0]
+                else:
+                    n_settings = 1
+                    settings_file_use = settings_file
+                if n_settings != n_files:
+                    raise ValueError(
+                        "Number of settings_files must match number of cal_files"
+                    )
+
+        if multi:
+            self.read(
+                filename[0],
+                file_type=file_type,
+                read_data=read_data,
+                use_future_array_shapes=use_future_array_shapes,
+                # checking parameters
+                run_check=run_check,
+                check_extra=check_extra,
+                run_check_acceptability=run_check_acceptability,
+                # file-type specific parameters
+                # FHD
+                obs_file=obs_file_use,
+                layout_file=layout_file_use,
+                settings_file=settings_file_use,
+                raw=raw,
+                extra_history=extra_history,
+            )
+            uv_list = []
+            for ind, file in enumerate(filename[1:]):
+                if file_type == "fhd":
+                    file_index = ind + 1
+                    obs_file_use = obs_file[file_index]
+                    if layout_file is not None:
+                        layout_file_use = layout_file[file_index]
+                    if settings_file is not None:
+                        settings_file_use = settings_file[file_index]
+
+                uvcal2 = UVCal()
+                uvcal2.read(
+                    file,
+                    read_data=read_data,
+                    file_type=file_type,
+                    use_future_array_shapes=use_future_array_shapes,
+                    # checking parameters
+                    run_check=run_check,
+                    check_extra=check_extra,
+                    run_check_acceptability=run_check_acceptability,
+                    # file-type specific parameters
+                    # FHD
+                    obs_file=obs_file_use,
+                    layout_file=layout_file_use,
+                    settings_file=settings_file_use,
+                    raw=raw,
+                    extra_history=extra_history,
+                )
+                uv_list.append(uvcal2)
+            # Concatenate once at end
+            if axis is not None:
+                # fast_concat to operates on lists
+                self.fast_concat(
+                    uv_list,
+                    axis,
+                    run_check=run_check,
+                    check_extra=check_extra,
+                    run_check_acceptability=run_check_acceptability,
+                    inplace=True,
+                )
+            else:
+                # Too much work to rewrite __add__ to operate on lists
+                # of files, so instead doing a binary tree merge
```

Comment:
Clever!

Reply:
I can't claim credit, this is flat-out stolen from work Aaron Parsons did on UVData.